### PR TITLE
Altair: new protos/state object

### PR DESCRIPTION
**Pull request by prestonvanloon**
**What type of PR is this?**

Other

**What does this PR do? Why is it needed?**

This PR brings the new data structures for the Altair state from `hf1` branch.

Please review the spec definition of the Beacon State after the Altair hard fork.

https://github.com/ethereum/eth2.0-specs/blob/master/specs/altair/beacon-chain.md#beaconstate

Approval on this PR means that the reviewer has reviewed that these changes are inline with the eth2 specification for Altair state, the code is clear and well documented, and the changes meet the high standards of the Prysm project.

**Which issues(s) does this PR fix?**

**Other notes for review**

Tracking: #8638

### CODE REVIEW

File: proto/beacon/p2p/v1/types.proto [hunk 1]
```diff
@@ -116,4 +163,18 @@ message DepositMessage {
...
\ No newline at end of file
+}
+
+// SyncCommittee serves as committees to facilitate light client syncing to beacon chain.
+message SyncCommittee {
+  repeated bytes pubkeys = 1  [(ethereum.eth.ext.ssz_size) = "512,48"];
+  bytes aggregate_pubkey = 2 [(ethereum.eth.ext.ssz_size) = "48"];
+}
+
+// SyncCommitteeSigningData to be used for hash tree root then sign by sync committee participants.
```
Comment (prestonvanloon) [inline proto/beacon/p2p/v1/types.proto:174]: @terencechain can this be made more clear?

File: proto/beacon/p2p/v1/types.proto [hunk 2]
```diff
@@ -116,4 +163,18 @@ message DepositMessage {
...
+message SyncCommitteeSigningData{
+  // Slot of this signing data.
+  uint64 slot = 1;
+  // Sub committee index of this signing data.
```
Comment (prestonvanloon) [inline proto/beacon/p2p/v1/types.proto:178]:
```suggestion
  // Subcommittee index of this signing data.
```
Is subcommittee one word?

File: proto/beacon/p2p/v1/types.proto (hunk 1 above)
Comment (terencechain) [reply proto/beacon/p2p/v1/types.proto:174]: `to be used to sign over then check whether the aggregator is selected within the subcommittee`

Also, the latest release changed the name. Do you mind update it?
```python
class SyncAggregatorSelectionData(Container):
    slot: Slot
    subcommittee_index: uint64
```
https://github.com/ethereum/eth2.0-specs/blob/dev/specs/altair/validator.md#syncaggregatorselectiondata

Comment (prestonvanloon) [reply proto/beacon/p2p/v1/types.proto:174]: Both of these sound good

**Review by terencechain — APPROVED** [review]

**Review by nisdas — APPROVED** [review]

**Pull request merged**